File: lib/src/main/java/net/kibotu/android/recyclerviewpresenter/OnItemFocusChangeListener.java
package net.kibotu.android.recyclerviewpresenter;

import android.view.View;

import androidx.annotation.NonNull;

/**
 * Created by <a href="https://about.me/janrabe">Jan Rabe</a>.
 */

public interface OnItemFocusChangeListener<T> {

    /**
     * Invokes click listener.
     *
     * @param item     Model of the adapter.
     * @param view     View that has been clicked.
     * @param hasFocus <code>True</code> if it is focused.
     */
    void onFocusChange(@NonNull final T item, @NonNull final View view, boolean hasFocus);
}

File: lib/src/main/java/net/kibotu/android/recyclerviewpresenter/v1/OnItemFocusChangeListener.kt
package net.kibotu.android.recyclerviewpresenter.v1

import android.view.View

/**
 * Created by [Jan Rabe](https://about.me/janrabe).
 */

interface OnItemFocusChangeListener<T> {

    /**
     * Invokes click listener.
     *
     * @param item     Model of the adapter.
     * @param view     View that has been clicked.
     * @param hasFocus `True` if it is focused.
     */
    fun onFocusChange(item: T, view: View, hasFocus: Boolean)
}